### PR TITLE
Task API Backend - Simple functionality

### PR DESCRIPTION
**Pull request by shelleychen318**
## Notion ticket link

[Task API Backend - Simple functionality](https://www.notion.so/uwblueprintexecs/F23-Task-Board-181d7a3269374b4c95cca79e331b87e1?p=dcb7a98834284520a3177a42ca6f2629&pm=s)

## Implementation description
* Created the interface for task API

## Steps to test
1.

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Comment by williamtran10** [comment]
Did you nest the `implementations` folder into the `interfaces` folder? I think its showing that those files were changed.

**Comment by shelleychen318** [comment]
> Did you nest the `implementations` folder into the `interfaces` folder? I think its showing that those files were changed.

oops i will fix that

File: backend/typescript/services/interfaces/taskService.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+     * @throws Error if task update fails
+     */
+    updateTaskById(taskId: string, task: UpdateTaskDTO): Promise<TaskDTO>;
+
+    /**
+     * Delete a task by id
+     * @param taskId task's taskId
+     * @throws Error if task deletion fails
+     */
+    deleteTaskById(taskId: string): Promise<void>;
```
Comment (williamtran10) [inline backend/typescript/services/interfaces/taskService.ts]: Delete should return the entity thats deleted which in this case is `TaskDTO`

File: backend/typescript/services/interfaces/taskService.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    createTask(task: CreateTaskDTO): Promise<TaskDTO>;
+    
+    /**
+     * Update a task.
+     * @param taskId task's id
+     * @param task the task to be updated
+     * @returns a TaskDTO with the updated task's information
+     * @throws Error if task update fails
+     */
+    updateTaskById(taskId: string, task: UpdateTaskDTO): Promise<TaskDTO>;
```
Comment (williamtran10) [inline backend/typescript/services/interfaces/taskService.ts]: Does `UpdateTaskDTO` need to be different from `TaskDTO`?

File: backend/typescript/services/interfaces/taskService.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
+import {
+    CreateTaskDTO,
+    Role,
+    SignUpMethod,
+    UpdateTaskDTO,
+    TaskDTO,
+  } from "../../types"; // not sure lmao
```
Comment (williamtran10) [inline backend/typescript/services/interfaces/taskService.ts]: You can create these types at the top of this file.

**Pull request closed without merging**